recipes/kokkos-kernels/meta.yaml

```diff
@@ -0,0 +1,60 @@
+{% set name = "kokkos-kernels" %}
+{% set version = "3.7.01" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/kokkos/kokkos-kernels/archive/refs/tags/{{ version }}.tar.gz
+  sha256: b2060f5894bdaf7f7d4793b90444fac260460cfa80595afcbcb955518864b446
+
+build:
+  skip: true  # [win]
+  number: 0
+  run_exports:
+    - {{ pin_subpackage('kokkos-kernels') }}
+
+requirements:
+  build:
+    - cmake >=3.21.1,<4
+    - {{ compiler('cxx') }}
+    - llvm-openmp  # [osx]
+    - libgomp      # [linux]
+    - ninja
+  host:
+    - kokkos
+  run:
+    - kokkos
+
+test:
+  commands:
+  # Lots more CMake files; just checking for one
+    - test -f $PREFIX/lib/cmake/KokkosKernels/KokkosKernelsConfig.cmake  # [unix]
+  # Testing explicitly for all shared libs, but not links
+    - test -f $PREFIX/lib/libkokkoskernels.so  # [linux]
+    - test -f $PREFIX/lib/libkokkoskernels.dylib  # [osx]
+  # There are lots more headers; only test for one.
+    - test -f $PREFIX/include/KokkosKernels_Handle.hpp  # [unix]
+  # There are no binary
```

Comment:
It seems to me one can compile some dummy code against `kokkos-kernels`. How hard is it to come up with one? If we can do a simple compilation test here, it'd be great. You'd be surprised how many issues a naive test like this could catch 🙂

Reply:
I guess we could try building the examples and tests, would that work?

Reply:
There is an error when compiling the tests, which I have reported [here](https://github.com/kokkos/kokkos-kernels/issues/1791).

Reply:
Hi @leofang , what's the best way to get that file if the file is in `kokkos-kernels/example/batched_solve`. Should I copy from `$SRC_DIR/../work`?

Reply:
Found my answer to the previous questions, but my test fails because the c++ compiler is not found in the test phase. Where does one find g++ for a test?

Reply:
Sorry for late reply, Vincent, have a long day... I think we just need to switch to `g++`.

Reply:
For sure, no worries. I think I've tried that, but let me try again.

Reply:
Yes you did it in [153418a](https://github.com/conda-forge/staged-recipes/pull/22345/commits/153418a048e8bdfb997ae3bc67e81472a521d8fc), but you didn't add the compiler until [1a59fdc](https://github.com/conda-forge/staged-recipes/pull/22345/commits/1a59fdc5b276be5c9f9b2a3fd2bf73260704bb5d) 🙂

Reply:
linux-64 passed, great work Vincent!

Reply:
Thanks for your feedback ;)

Reply:
All passed!